server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -53,6 +54,7 @@
     private final BigArrays bigArrays;
     private final IntConsumer multiBucketConsumer;
     private IntArray docCounts;
```

Comment:
Should we switch to long now that a single document can add more than 1 ?
That's probably a follow up though, no need to change in this PR.

Reply:
Fixed in this PR